### PR TITLE
MINOR: Extract some codeblocks as methods to simplify readability

### PR DESCRIPTION
**Pull request by overpathz**
Small changes related to increase code readability by extracting some code blocks into separate methods.

Does not require any testing here.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/java/kafka/server/TierStateMachine.java [hunk 1]
```diff
@@ -229,12 +229,8 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
...
 
         RemoteLogSegmentMetadata remoteLogSegmentMetadata = rlm.fetchRemoteLogSegmentMetadata(topicPartition, targetEpoch, previousOffsetToLeaderLocalLogStartOffset)
-                .orElseThrow(() -> new RemoteStorageException("Couldn't build the state from remote store for partition: " + topicPartition +
-                        ", currentLeaderEpoch: " + currentLeaderEpoch +
-                        ", leaderLocalLogStartOffset: " + leaderLocalLogStartOffset +
-                        ", leaderLogStartOffset: " + leaderLogStartOffset +
-                        ", epoch: " + targetEpoch +
-                        "as the previous remote log segment metadata was not found"));
+                .orElseThrow(() -> buildRemoteStorageException(topicPartition, targetEpoch, currentLeaderEpoch,
+                        leaderLocalLogStartOffset, previousOffsetToLeaderLocalLogStartOffset));
```
Comment (AndrewJSchofield) [inline core/src/main/java/kafka/server/TierStateMachine.java]: It seems to me that the final argument here should be `leaderLogStartOffset` to match the previous behaviour.

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]

File: core/src/main/java/kafka/server/TierStateMachine.java (hunk 1 above)
Comment (overpathz) [reply core/src/main/java/kafka/server/TierStateMachine.java]: Thx. Addressed in the recent commit.

Comment (overpathz) [reply core/src/main/java/kafka/server/TierStateMachine.java]: @AndrewJSchofield Additional ping for possible notification miss

**Review by AndrewJSchofield — APPROVED** [review]

**Comment by AndrewJSchofield** [comment]
@overpathz Thanks for the PR. lgtm. Please resolve the conflicts.

**Review by AndrewJSchofield — APPROVED** [review]

**Pull request merged**